pkg/lib/cockpit.d.ts

```diff
@@ -94,6 +94,49 @@ declare module 'cockpit' {
     function channel(options: ChannelOptions & { binary?: false; }): Channel<string>;
     function channel(options: ChannelOptions & { binary: true; }): Channel<Uint8Array>;
 
+    /* === cockpit.spawn ============================= */
+
+    interface DeferredPromise<T> extends Promise<T> {
+        /* jQuery Promise compatibility */
+        done(callback: (data: T) => void): DeferredPromise<T>
+        fail(callback: (exc: Error) => void): DeferredPromise<T>
+        always(callback: () => void): DeferredPromise<T>
+        progress(callback: (message: T, cancel: () => void) => void): DeferredPromise<T>
+    }
+
+    interface Deferred<T> {
```

Comment:
I'd be interested in seeing if you had any kind of success porting `test-promise` to `.ts` based on this.  It's the smallest qunit test case.  I looked at it but decided to wait until these changes...